prover/src/lib.rs

```diff
@@ -328,17 +320,18 @@ pub trait Prover {
         // evaluate constraints specified by the AIR over the constraint evaluation domain, and
         // compute random linear combinations of these evaluations using coefficients drawn from
         // the channel
-        #[cfg(feature = "std")]
-        let now = Instant::now();
-        let constraint_coeffs = channel.get_constraint_composition_coeffs();
-        let evaluator = self.new_evaluator(&air, aux_trace_rand_elements, constraint_coeffs);
-        let composition_poly_trace = evaluator.evaluate(&trace_lde, &domain);
-        #[cfg(feature = "std")]
-        debug!(
-            "Evaluated constraints over domain of 2^{} elements in {} ms",
-            composition_poly_trace.num_rows().ilog2(),
-            now.elapsed().as_millis()
-        );
+        let ce_domain_size = air.ce_domain_size();
+        let composition_poly_trace = {
+            let _ = info_span!("evaluate_constraints", ce_domain_size).entered();
+            let evaluator = self.new_evaluator(
+                &air,
+                aux_trace_rand_elements,
+                channel.get_constraint_composition_coeffs(),
+            );
+
+            evaluator.evaluate(&trace_lde, &domain)
+        };
+        assert_eq!(composition_poly_trace.num_rows(), ce_domain_size);
 
         // 3 ----- commit to constraint evaluations -----------------------------------------------
 
```

Comment:
Similar to the previous comments, let's create a span for this step as well. We can call it `commit_to_constraint_evaluations`.